### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by InFTord**
Make sure you familiarize yourself with our contributing guidelines.

# Summary
Adds memberpage plugin. Updates our workflow to work properly. Moves docs folder to DSharpPlus.Docs

# Notes
h

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -4,13 +4,16 @@ on:
...
+      - 'DSharpPlus.Docs/**'
   release:
     types:
       - published
+  workflow_dispatch:
+  
 env:
   DOTNET_NOLOGO: true
   DOTNET_CLI_TELEMETRY_OPTOUT: true
+  DOTNET_INSTALL_DIR: "~/.dotnet"
```
Comment (OoLunar) [inline .github/workflows/docs.yml:16]: Why are we changing the install dir?

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -19,19 +22,17 @@ jobs:
     steps:
     - uses: actions/checkout@v3
     - name: Setup .NET
-      uses: actions/setup-dotnet@v3
+      uses: actions/setup-dotnet@v3.0.3
```
Comment (OoLunar) [inline .github/workflows/docs.yml:25]: Why are we specifying a specific version?

File: DSharpPlus.Docs/DSharpPlus.docs.csproj [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (OoLunar) [inline DSharpPlus.Docs/DSharpPlus.docs.csproj:1]: I personally believe that this should be put in the `tools/` directory

File: .github/workflows/docs.yml (hunk 1 above)
Comment (InFTord) [reply .github/workflows/docs.yml:16]: my runner was dying because of dir being /usr/share/dotnet

File: .github/workflows/docs.yml (hunk 2 above)
Comment (InFTord) [reply .github/workflows/docs.yml:25]: i-
i have no idea why i did that

File: DSharpPlus.Docs/DSharpPlus.docs.csproj (hunk 3 above)
Comment (InFTord) [reply DSharpPlus.Docs/DSharpPlus.docs.csproj:1]: maybe

**Comment by OoLunar** [comment]
This PR has been tested and is ready for merge.

My only complaint is that the API reference is blank. I feel like it should be littered with references to types on the sidebar instead. 
image

**Comment by OoLunar** [comment]
Oh nevermind, I think the types are just missing in general. The reason why the references aren't working in the FAQ is because there isn't anything to reference in the first place; it simply wasn't generated.

**Comment by OoLunar** [comment]
Taking a closer look, this only seems to happen when using the `DSharpPlus.Docs` tool to generate the docs, not when using the CLI `docfx` tool. Likely a configuration issue with our program then.

**Comment by InFTord** [comment]
> This PR has been tested and is ready for merge.
> 
> My only complaint is that the API reference is blank. I feel like it should be littered with references to types on the sidebar instead. 

Fixed in latest commit.

**Comment by OoLunar** [comment]
> > This PR has been tested and is ready for merge.
> > My only complaint is that the API reference is blank. I feel like it should be littered with references to types on the sidebar instead.
> 
> Fixed in latest commit.

Unable to reproduce:

```
git clean -fdx \
  && docfx docs/docfx.json \
  && cd tools/Docs \
  && dotnet run \
  && cd ../../docs/_site \
  && python -m http.server
```

image

**Comment by OoLunar** [comment]
Nevermind forgot to `dotnet build`

**Pull request merged**